### PR TITLE
HEEDLS-225 do not add session for invalid course

### PR DESCRIPTION
**Pull request by benjimarshall**
## Changes
- Fix session adding bug for invalid course
- Add tests to cover this

## Testing
Add and modify the unit tests, and run them all.

### CODE REVIEW

**Review by FridaTveit — COMMENTED** [review]
A few of the comments on your other PR also apply here, in particular: 
* Not using "missing progress" in the test names
* Redirecting to 404 instead of 500 if unable to create progress (feel free to do in the other PR if you prefer but it's probably less likely to create conflicts if you do it here)

Otherwise looks good 🙂

**Comment by benjimarshall** [comment]
Done both of those bits in 89bdba6 🙂

File: DigitalLearningSolutions.Web/Controllers/LearningMenuController/LearningMenuController.cs [hunk 1]
```diff
@@ -51,9 +51,9 @@ public IActionResult Index(int customisationId)
             if (progressId == null)
             {
                 logger.LogError(
-                    "Redirecting to 500 as no progress id was returned. " +
+                    "Redirecting to 404 as no progress id was returned. " +
                     $"Candidate id: {candidateId}, customisation id: {customisationId}, centre id: {centreId}");
-                return RedirectToAction("StatusCode", "LearningSolutions", new { code = 500 });
+                return RedirectToAction("StatusCode", "LearningSolutions", new { code = 404 });
```
Comment (FridaTveit) [inline DigitalLearningSolutions.Web/Controllers/LearningMenuController/LearningMenuController.cs:56]: Do we not have a test for this which needs updating? If we don't, could you add one? 🙂

**Review by FridaTveit — COMMENTED** [review]
One comment, otherwise looks good 🙂

File: DigitalLearningSolutions.Web/Controllers/LearningMenuController/LearningMenuController.cs (hunk 1 above)
Comment (benjimarshall) [reply DigitalLearningSolutions.Web/Controllers/LearningMenuController/LearningMenuController.cs:56]: Yep, added in 149fac0 🙂

**Review by FridaTveit — APPROVED** [review]
Looks good 🙂

**Pull request merged**